samples/common/src/utils.rs

```diff
@@ -205,7 +234,40 @@ pub async fn retrieve_invehicle_digital_twin_uri(
         }
     };
 
-    Ok(result)
+    get_uri(&result).map_err(|err| {
+        format!("Failed to retrieve the in-vehicle digital twin service's URI due to error: {err}")
```

Comment:
Are you sure that you need the .to_string() call?  See: https://doc.rust-lang.org/std/macro.format.html

Reply:
good catch, removed